Sage/carbon
# refactor(field-help): convert files to typescript

### Proposed behaviour


Converts files for `FieldHelp` to TypeScript
### Current behaviour


Files for `FieldHelp` are JavaScript

### Checklist




#### QA


### Additional context



### Testing instructions


Linked sandbox and regression test of inputs

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.


https://codesandbox.io/s/carbon-quickstart-typescript-forked-kitc96?file=/src/index.tsx -- some inputs that use FieldHelp, you may need to export and run locally
